internal/errors/errors.go

```diff
@@ -0,0 +1,52 @@
+// +build !debug
+
+// Package errors provides a simple API to create and compare errors. A debug version of this package
+// exists, but captures stacktraces when error are created or wrapped. It is accessible through the
+// the "debug" build tag.
+package errors
+
+import (
+	baseErrors "errors"
+
+	"github.com/genjidb/genji/internal/stringutil"
+)
+
+// New takes either a string, an existing error and returns a standard error. If the error is nil,
+// it returns nil, enabling to wrap functions that returns an error directly.
+// If a string is passed, a new error is created based on that string.
+// If any other type is passed, it will panic.
+func New(e interface{}) error {
```

Comment:
If think this function should be called `Wrap` instead of `New`:
- In both build modes, it doesn't actually create a new error in all cases.
- It mixes creating an error out of a string and returning / wrapping an existing error.
- Its counterpart is named `Unwrap`
- When it's used with an existing an error, it feels weird to read `errors.New(e)`

This is a handy function but it mixes two needs that could be split in more explicit functions:

- wrapping an error

```go
func Wrap(e error) error {
  if e == nil { return nil }

 // no-debug
  return stringutil.Errorf("%w", e)
 // debug
  // stacktrace et autres joyeusetés...
}
```

- creating a function out of a string:
```go
func New(e string) error {
  // no-debug
  return errors.New(e)
// debug
  return Wrap(errors.New(e))
}
```

Reply:
I strongly hesitated on this one, though I entirely agree that `New(error)` feels a bit weird albeit it's quite handy.  I'll change it.